dist/common_functions.py

```diff
@@ -30,6 +29,40 @@ def last_commit_from_dev():
     return subprocess.run(f"git rev-parse {commit_on_dev}",
         shell=True, capture_output=True, text=True).stdout.strip()
 
+
+def is_fast_mode(fast = None):
+    fast = fast or "-F" in sys.argv[1:] or os.environ.get('FAST', '') != ''
+    if fast:
+        os.environ["FAST"] = "1"
+    return fast
+
+
+def files_changed_from_commit(prefix, commit = None):
+    # `git diff --name-only` returns file names relative to the root of the repository.
+    # If we want to use these file names from another location, we need to prefix them.
+    if commit is None:
+        commit = last_commit_from_dev()
+    # Find the files that have changed since the last commit
+    return [prefix+f for f in subprocess.run(f"git diff --name-only {commit}",
+        shell=True, capture_output=True, text=True).stdout.strip().splitlines()]
+
+
+def filter_changed_files(in_generator, prefix, commit = None):
```

Comment:
If the `prefix` name needs a comment, then it's probably not obvious enough - what about `relative_path`, or `rpath`, or some similar variant?

Reply:
The issue is that `git diff --name-only` gives the names relative to the repo root but there are infinite ways to refer to these files including "dist/../{path}" and many others. All of these will not work even though they are technically valid relative paths.
This prefix must exactly match the way the script refers to the root directory when looking for files.

This works only because our scripts currently use either "" or "../" as prefix. To make it robust and error prone, I need to call an "abspath" on each file. But I don't want to make it so much heavier than it's required.

There must be a better way to put it.

I'd like to avoid tricking developers into thinking about it as a relative path and assuming it to have other relative path's properties. I've put an extra comment into the source. Please see it if describes the thing well.

Reply:
Oh, thanks. Yes, I think `prefix` is a reasonable name here, and the comment helps a lot. That could've been very frustrating to debug without the explanation.